apps/legacy/tools/history/issue.py

```diff
@@ -152,12 +187,21 @@ def migrate(klass, anon):
                     # Same story, but a different user, and at least one of
                     # the current objects needs saving.  So save under the old
                     # user and mark all stories as clean.
-                    klass.gather_changeset(last_i, attached_stories, anon)
-
+                    if not use_earlier_data:
```

Comment:
This block of code appears 10 times, and I *think* it's the same each time?  A function (or just modifying gather_changeset() to handle this) would make all of this a bit easier to follow.